### PR TITLE
Mongo errors 624

### LINKED ISSUE #624: Momo generates a lot of Mongo Errors
Looks like we expect and gracefully handle a Mongo error here: https://github.com/TNG/momo-scheduler/blob/main/src/repository/SchedulesRepository.ts#L57

However, this spams the logs of any responsible DB with errors like `MongoServerError: E11000 duplicate key error collection:(...)`.

There are expected at this place and we do handle them correctly but the behaviour can make people think there's something wrong with the service. 

We should investigate if there's a different way to perform this check

### PR DESCRIPTION
**Pull request by weissu42**
resolves #624

### CODE REVIEW

File: src/repository/SchedulesRepository.ts [hunk 1]
```diff
@@ -25,97 +27,127 @@ export class SchedulesRepository extends Repository<ScheduleEntity> {
...
-            lastAlive,
-            executions: {},
-          },
-        },
-        {
-          upsert: true,
-          returnDocument: 'after',
-        },
+      await this.collection.updateOne(
+        { $or: [deadSchedule, thisSchedule] },
```
Comment (YaniKolev) [inline src/repository/SchedulesRepository.ts:80]: I don't quite get this $or clause.

File: src/schedule/SchedulePing.ts [hunk 2]
```diff
@@ -30,33 +30,42 @@ export class SchedulePing {
...
-        this.logger.debug('Finished starting scheduled jobs');
-      }
+    const now = DateTime.now().toMillis();
+    const active = await this.schedulesRepository.isActiveSchedule(now);
+    if (!active) {
+      this.logger.debug('This schedule is not active', this.schedulesRepository.getLogData());
+      return;
+    }
+
+    if (!(await this.schedulesRepository.setActiveSchedule(now))) {
```
Comment (YaniKolev) [inline src/schedule/SchedulePing.ts:46]: Can we extract the condition to a variable for documentation purposes?
```suggestion
    const somethingThatDescribesWhatIsReturned = await this.schedulesRepository.setActiveSchedule(now)
    if (!somethingThatDescribesWhatIsReturned) {
```

**Review by YaniKolev — APPROVED** [review]

File: src/repository/SchedulesRepository.ts (hunk 1 above)
Comment (weissu42) [reply src/repository/SchedulesRepository.ts:85]: We want to update in exactly two cases:
there is an active schedule, but it is dead and will be replaced by me
or: I am the active schedule currently and only will update the timestamp

Any hints how to make this clearer are welcome. Maybe I can add some words to the jsdoc.

Comment (weissu42) [reply src/repository/SchedulesRepository.ts:99]: I added a comment and extended the jsdoc of the method. Is it clearer now?

File: src/schedule/SchedulePing.ts (hunk 2 above)
Comment (weissu42) [reply src/schedule/SchedulePing.ts]: I refactored this again, now calling the "isActiveSchedule" inside the "setActiveSchedule" in SchedulesRepository, making the code in SchedulePing clearer.

File: src/repository/SchedulesRepository.ts (hunk 1 above)
Comment (YaniKolev) [reply src/repository/SchedulesRepository.ts:99]: Awesome, thanks! Looks good!

**Review by YaniKolev — APPROVED** [review]

**Pull request merged**